Backend/src/main/java/ca/mcgill/cooperator/controller/StudentController.java

```diff
@@ -40,7 +48,48 @@
     @GetMapping("")
     public List<StudentDto> getAllStudents() {
         List<Student> s = studentService.getAllStudents();
-        return ControllerUtils.convertToDto(s);
+        return ControllerUtils.convertStudentListToDto(s);
+    }
+    
+    /**
+     * Get students with filters
+     *
+     * @return List<StudentDto>
+     */
+    @GetMapping("/filter")
+    public Set<StudentDto> getStudentFiltered(
+    		@RequestParam Season season, 
+    		@RequestParam Integer year, 
+    		@RequestParam String name, 
+    		@RequestParam CoopStatus status) {
+    	Set<CourseOffering> result = courseOfferingService.getAllCourseOfferingsSet();
+    	if(year != null) {
```

Comment:
You can combine this with the method above and make the params optional; if they're not present then return all students, otherwise do the filtering (this is better design from an external API perspective so that we have less routes)

Reply:
i dont think this is that good of an idea now that im trying to implement bc it means i have to check to make sure that all the filter inputs were null/not there to return students which is a whole other portion of the method with just a huge if statement 

i thought at first it would just be if the size of the result is 0 then just return all but thats wrong bc the combo of filters could just return no matching records soooo i can combine them but its not as slick as i originally thought

Reply:
<img width="795" alt="image" src="https://user-images.githubusercontent.com/35476226/77828861-6d853900-70f4-11ea-980b-97a3c4daa9c3.png">

IDC i can do this if you want

Reply:
ok im kidding it doesnt work when its this way

Reply:
Why doesn't it work? The code looks ok as far as I can tell

Reply:
Lol it turns out there was a bug in the filter and it was EXPOSED when i added that line but all good i fixed it

Reply:
Ok nice I would say keep it this way as one route then